### ISSUE TITLE
Missing autoIncrement: true

### ISSUE THREAD

**Issue by KThen** [sequelize/sequelize-auto#237]
**Issue:**
`autoIncrement: true` is missing on generated models.

**Environment:**
OS: macOS High Sierra
MySQL: 8.0.11
node: v8.11.1
npm: v6.1.0
sequelize-auto: 0.4.29

**Details:**
During generation I noticed that the EXTRA column that is required is in uppercase.

**Code reference:**
https://github.com/sequelize/sequelize-auto/blob/c16e8c0b5b977372be0209bde1bd22c920538065/lib/dialects/index.js#L92

**Suggestions:**
`return _.isObject(record) && (_.has(record, 'extra') || _.has(record, 'EXTRA')) && (record.extra === "auto_increment" || record.EXTRA === "auto_increment")`

**Comment by aruntk** [comment]
until this is fixed you can use [@tkay/sequelize-auto](https://www.npmjs.com/package/@tkay/sequelize-auto)

**Comment by ilantc** [comment]
This is because `sequelize-auto` uses its own query to grab the keys instead of using sequelize's and it does:

```

  getForeignKeysQuery: function(tableName, schemaName) {
    return "SELECT \
        K.CONSTRAINT_NAME as constraint_name \
      , K.CONSTRAINT_SCHEMA as source_schema \
      , K.TABLE_SCHEMA as source_table \
      , K.COLUMN_NAME as source_column \
      , K.REFERENCED_TABLE_SCHEMA AS target_schema \
      , K.REFERENCED_TABLE_NAME AS target_table \
      , K.REFERENCED_COLUMN_NAME AS target_column \
      , C.extra \
      , C.COLUMN_KEY AS column_key \
      FROM INFORMATION_SCHEMA.KEY_COLUMN_USAGE AS K \
      LEFT JOIN INFORMATION_SCHEMA.COLUMNS AS C \
        ON C.TABLE_NAME = K.TABLE_NAME AND C.COLUMN_NAME = K.COLUMN_NAME \
      WHERE \
        K.TABLE_NAME = '" + tableName + "' \
        AND K.CONSTRAINT_SCHEMA = '" + schemaName + "';";
  },
```

Changing `C.extra` to `C.extra as extra` should fix the issue.

I can PR this if you'd like

**Comment by FredLackeyOfficial** [comment]
Take a look at [`file-line-replacer`](https://github.com/FredLackey/file-line-replacer) when you have a moment.  I've had the same problem for years and cannot stand dealing with writing scripts to tweak the final files.  I created [`file-line-replacer`](https://github.com/FredLackey/file-line-replacer) to make fixing these files easier.  This allows sequelize-auto to be used, without issue, and then adjust the files when you're finished.

Here are the steps I use.  First, install it (I prefer installing it globally)...

`npm i -g file-line-replacer`

... and then, from inside of the project using sequelize, I run ... 

```
file-line-replacer \
  --search-dir "/Users/flackey/my-project/src/data/models" \
  --backup-dir "/Users/flackey/my-project/_backup" \
  --old-lines "allowNull: false,|primaryKey: true" \
  --new-lines "autoIncrement: true,|primaryKey: true" \
  --overwrite
``` 
The above command will tweak your models, adding the `autoIncrement` key, and create a backup for each file before it is modified.

**Comment by steveschmitt** [comment]
Fixed in `sequelize-auto@0.5.0`
